app/src/forms/auth/service.js

```diff
@@ -212,8 +214,41 @@ const service = {
     const forms = await service.getUserForms(user, params); // get forms for user (filtered by params)...
     params.active = false;
     const deletedForms = await service.getUserForms(user, params); // get forms for user (filtered by params)...
-    return Object.assign({}, {...user, forms: forms, deletedForms: deletedForms});
+    return Object.assign({}, { ...user, forms: forms, deletedForms: deletedForms });
+  },
+
+  // -------------------------------------------------------------------------------------------------------------
+  // Submission Data
+  // -------------------------------------------------------------------------------------------------------------
+  checkSubmissionPermission: async (currentUser, submissionId, permissions) => {
+    if (currentUser.public) return false;
+    return FormSubmissionUserPermissions.query()
+      .modify('filterSubmissionId', submissionId)
+      .modify('filterUserId', currentUser.id)
+      .modify('filterByPermissions', permissions)
+      .first();
+  },
```

Comment:
Is this query missing a `throwIfNotFound()` in the chain?

Reply:
No this one is supposed to return which permission intersections you have for the supplied id, user, permission list. So the array of permissions if so, or falsey if nothing.